### PR TITLE
Extend gateway tests to validate connectivity from gateway to pod

### PR DESCRIPTION
**Pull request by fedepaol**
**- What this PR does and why is it needed**

We currently validate connectivity with various protocols / ip families
from pod to egress. Here we validate the same path but starting from the
gateway containers that mimic the gw pod.

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: test/e2e/external_gateways.go [hunk 1]
```diff
@@ -18,7 +20,11 @@ import (
 
 // This is the image used for the containers acting as externalgateways, built
 // out from the e2e/images/Dockerfile.frr dockerfile
-const externalContainerImage = "quay.io/fpaoline/ovnkbfdtest:0.2"
+const (
+	externalContainerImage = "quay.io/fpaoline/ovnkbfdtest:0.2"
```
Comment (astoycos) [inline test/e2e/external_gateways.go:24]: Is there not a better repo for housing OVN-K CI images

Comment (fedepaol) [reply test/e2e/external_gateways.go:24]: This came out when I created the original one (now I am only adding a new constant here) and we agreed with @trozet that it was fine this way. I am providing the dockerfile here so it should be easy to rebuild it if my quay repo dies for some reason.

**Comment by astoycos** [comment]
That build failure has been popping up in our CI recently...

**Review by trozet — APPROVED** [review]
/lgtm

**Pull request merged**